### DISCUSSION TITLE
Text getting cut off in notification

### DISCUSSION THREAD

**Discussion by sgexcit** [Windos/BurntToast#156] [Q&A]
I'm building a notification that is using three New-BTText items. The issue is without explanation it'll cut of sections of the text, or not show it at all. I've been able to side-step this a little with the New-BTHeader function (which creates some problems of it's own with text format). Is this a known issue? I tried using -MaxLines with New-BTText to force it, but that doesn't appear to do anything.

**Comment by Windos** [comment]
Hey @sgexcit, can you post some dummy examples of what you're trying to achieve?

**Reply by sgexcit** [reply]
Certainly!

Here is the example code (assuming BurntToast already installed and imported):

```
$AppID = (Get-StartApps | Where-Object {$_.name -eq "ApplicationXyz"} | Select "AppID").AppID
if ($AppID -eq $null)
{
    $AppID = (Get-StartApps | Where-Object {$_.name -eq "Settings"} | Select "AppID").AppID
}
$header1 = New-BTHeader -Title "Message from Company ABC"
$text1 = New-BTText -Text "Here is some information about this message. Here is some additional information about this message."
$text2 = New-BTText -Text "-"
$text3 = New-BTText -Text "IMPORTANT: Here is some really important information. This is about the length of the actual application message I am trying to send."
$button1 = New-BTButton -Content "Snooze" -Snooze
$button2 = New-BTButton -Content "Dismiss" -Dismiss
$action = New-BTAction -Buttons $button1, $button2
$binding = New-BTBinding -Children $text1, $text2, $text3
$visual = New-BTVisual -BindingGeneric $binding
$content = New-BTContent -Header $header1 -Visual $visual -Actions $action
Submit-BTNotification -AppID $AppID -Content $Content
```

This is what it outputs, note how it cuts off the ends of the sentences, and the Header bolds $text1 for some reason. What am I doing wrong?

![image](https://user-images.githubusercontent.com/76213604/109701889-8c404b00-7b61-11eb-9b27-5d5cba48c544.png)

**Reply by Windos** [reply]
This is going to go a bit into the weeds (Sorry! Just working through a thought process, will likely inform a blog post and a topic for documentation.)

We're stuck to working within the framework for these notifications provided to us by Microsoft. These text elements currently have the following caveats (as per the [MS Docs](https://docs.microsoft.com/en-us/windows/uwp/design/shell/tiles-and-notifications/adaptive-interactive-toasts?tabs=builder-syntax#text-elements)):

+ You **must** have one text element
+ You _can_ have two additional text elements, for a total of three
+ The first element is always a "title", it is shown bold and can be up to 2 lines
+ The other two text elements can use a _total_ of 4 lines **combined**. This could be 3 lines for element 2 and 1 for element 3, or any other combination including element 2 consuming all 4 available lines.

![Text element explainer](https://user-images.githubusercontent.com/6955786/109711104-5bb0ed80-7c03-11eb-90f5-a40a883d0942.png)

You'll notice my screenshot does a decent job of actually showing the third and largest text element. This is just a difference in Windows version, I'm running 20H2 which changed how toasts display. It's no silver bullet, it just happens that the text is slightly smaller so it can fit more onto three lines. Anything from adding more text to the user increasing their desktop scaling could force the text out of sight.

-----

As an aside, the [header](https://docs.microsoft.com/en-us/windows/uwp/design/shell/tiles-and-notifications/adaptive-interactive-toasts?tabs=builder-syntax#headers) (via `New-BTHeader`) is its own thing and is not included in these "text elements". The actual purpose of the headers is for categorization in the Action Center.

![Headers Grouping](https://user-images.githubusercontent.com/6955786/109710562-ae3dda00-7c02-11eb-9c40-87fe4d8a90e0.png)

-----

One thing you could look to do is replace the line with the "-" with the second half of your title:

![attempt 1](https://user-images.githubusercontent.com/6955786/109711944-4be5d900-7c04-11eb-8af9-e83a7add834c.png)

But as you can see, I'm now overflowing my four-line allocation... so no good!

My actual solution is to use a newish feature of BurntToast: columns (or "[AdaptiveGroups](https://docs.microsoft.com/en-us/windows/uwp/design/shell/tiles-and-notifications/adaptive-interactive-toasts?tabs=builder-syntax#adaptive-content)" in the MS Docs)

```powershell
$header1 = New-BTHeader -Title "Message from Company ABC"
$text1 = New-BTText -Text "Here is some information about this message."
$text2 = New-BTText -Text "Here is some additional information about this message."

$text3 = New-BTText -Text "IMPORTANT: Here is some really important information. This is about the length of the actual application message I am trying to send." -Wrap
$column1 = New-BTColumn -Children $text3

$button1 = New-BTButton -Content "Snooze" -Snooze
$button2 = New-BTButton -Content "Dismiss" -Dismiss
$action = New-BTAction -Buttons $button1, $button2
$binding = New-BTBinding -Children $text1, $text2 -Column $column1
$visual = New-BTVisual -BindingGeneric $binding
$content = New-BTContent -Header $header1 -Visual $visual -Actions $action
Submit-BTNotification -Content $Content
```

![Column example](https://user-images.githubusercontent.com/6955786/109712550-070e7200-7c05-11eb-8d7c-985a5413b90e.png)

These columns are great for showing detailed data, I've got two demos of their use: [Music Ticker](https://github.com/Windos/Talks/blob/main/2021/02%20-%20February/23%20-%20DuPSUG%20-%20Croque%20Monsieur/Demos/10_Columns.ps1) & [Weather](https://github.com/Windos/Talks/blob/main/2021/02%20-%20February/23%20-%20DuPSUG%20-%20Croque%20Monsieur/Demos/11_Weather.ps1).

Here, however, we've just doing a single column and dumping the text into it.

Note the `-Wrap` switch, by default text in a "column" will only use one line, that makes to wrap to multiples.

You can also control the style of the text here:

```powershell
$text3 = New-BTText -Text "Column text example." -Wrap -Style Header
```

![Extreme example](https://user-images.githubusercontent.com/6955786/109712968-88fe9b00-7c05-11eb-833b-769f8b2ad031.png)

Yes, that's a very extreme example, but there's a lot of sane styles available via the `-Style` paramater.

n.b. that style thing is available on all text, but it only takes effect inside these columns, not the main body of the toast.

-----

Sorry that was so long winded, hope it helped?

**Reply by sgexcit** [reply]
Please don't apologize, I'm extremely happy you took the time to explain all that. I think this column feature in conjunction with the header should do exactly what I need, or at the very least give me the breathing room to input the information I need at bare minimum.

This has been monumentally assistive. I'm going to dig into this and if I have any other issues I'll be sure to reply to this thread.

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
